### DISCUSSION TITLE
Is there a way to install without TCC?

### DISCUSSION THREAD

**Discussion by yerlaser** [vlang/v#17951] [Questions and Answers] [answered]
Hi,
I already have a fully functional GCC/G++ installation.
When building V from sources, can I supply a parameter to avoid download and installation of TCC?

**Comment by JalonSolov** [comment, marked as answer]
tcc is downloaded and installed by default because it is much, much faster compiling that gcc/clang.

This is important for development because you get much faster turnaround while developing your code.

However, if you can stand to put up with the very noticeably slower compile times, you can set the env var `VFLAGS='-cc gcc'` and used `make -gcc` on install.

tcc will still be downloaded, but never used.

**Reply by yerlaser** [reply]
Yeah, I was aware of that flag. But, still would have an option to disable the download.
Thanks for the answer, nevertheless.

P.S. By the way, do you think it's worth it to create a Feature Request for that?

**Reply by JalonSolov** [reply]
The entire thirdparty/tcc dir is only 24MB on disk, so it's not taking up much space.  Also, when the `-native` backend is improved to the point that it can handle all of V, tcc will no longer be included.  V will still support a C backend, it just won't be used by default, so there won't be a need for a fast C compiler for dev.

I guess it depends on how much that 24MB bothers you as to whether or not you want to put in the feature request.

**Reply by yerlaser** [reply]
It's not the size actually. It's just to avoid extra third party dependency. Too many languages require huge external downloads without even warning users.

**Comment by Wajinn** [comment]
> Hi, I already have a fully functional GCC/G++ installation. When building V from sources, can I supply a parameter to avoid download and installation of TCC?

Your setup is not what everybody has.  There can be many new users of V who don't have GCC or other C compilers installed.  TCC is very small, quick, and useful in such circumstances.

> But, still would have an option to disable the download.

I do agree with this point, where TCC could be an optional download.  But of the opinion that TCC should be a default, and there being a flag to not download, for those that don't want it.  Even after the V backend has become mature, various people can like or need a small and fast C compiler (as TCC is), as V will still have its C backend.

**Comment by spytheman** [comment]
> can I supply a parameter to avoid download and installation of TCC?

Currently, no.
